samples/RecommendationsGatewayModule/RecommendationsGatewayModule.Core/Schemas/ProductRecommendationType.cs

```diff
@@ -36,7 +36,7 @@ public class ProductRecommendationType : ObjectGraphType<ProductRecommendation>
                     var loader = dataLoader.Context.GetOrAddBatchLoader<string, ExpProduct>($"recommendedProducts", (ids) => LoadProductsAsync(mediator, ids, includeFields));
 
                     // IMPORTANT: In order to avoid deadlocking on the loader we use the following construct (next 2 lines):
-                    var loadHandle = loader.LoadAsync(context.Source.ProductId);
+                    var loadHandle = loader.LoadAsync(context.Source.ProductId).GetResultAsync();
```

Comment:
Can you split this line into two to prevent possible deadlocks?

Reply:
done